### ISSUE TITLE
Unicode Error when launching PSI4 driver (using Windows)

### ISSUE THREAD

**Issue by cpossel** [Qiskit/qiskit-nature#260]
When using psi4 as driver for my vqe calculations, I get the following error:
```
Traceback (most recent call last):
  File "C:\Users\poc\Anaconda3\envs\qiskit-0-27-psi4-jupyter\Scripts\psi4-script.py", line 336, in <module>
    exec(content)

  File "<string>", line 31
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape

Printing out the relevant lines from the Psithon --> Python processed input file:


!----------------------------------------------------------------------------------!
!                                                                                  !
!  (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3:       !
!     truncated \UXXXXXXXX escape                                                  !
!                                                                                  !
!----------------------------------------------------------------------------------!

Traceback (most recent call last):
  File "c:\Users\poc\Documents\qiskit_scripts\groundstatesolver.py", line 336, in <module>
    compare_mappers()
  File "c:\Users\poc\Documents\qiskit_scripts\groundstatesolver.py", line 206, in compare_mappers
    second_q_op = es_problem.second_q_ops()
  File "C:\Users\poc\Anaconda3\envs\qiskit-0-27-psi4-jupyter\lib\site-packages\qiskit_nature\problems\second_quantization\electronic\electronic_structure_problem.py", line 64, in second_q_ops
    self._molecule_data = cast(QMolecule, self.driver.run())
  File "C:\Users\poc\Anaconda3\envs\qiskit-0-27-psi4-jupyter\lib\site-packages\qiskit_nature\drivers\psi4d\psi4driver.py", line 160, in run
    PSI4Driver._run_psi4(input_file, output_file)
  File "C:\Users\poc\Anaconda3\envs\qiskit-0-27-psi4-jupyter\lib\site-packages\qiskit_nature\drivers\psi4d\psi4driver.py", line 215, in _run_psi4
    raise QiskitNatureError('{} process return code {}\n{}'.format(
qiskit_nature.exceptions.QiskitNatureError: 'psi4 process return code 1\nTraceback (most recent call last):\n  File "C:\\Users\\poc\\Anaconda3\\envs\\qiskit-0-27-psi4-jupyter\\Scripts\\psi4-script.py", line 336, in <module>\n    exec(content)\n\n  File "<string>", line 31\nSyntaxError: (unicode error) \'unicodeescape\' codec can\'t decode bytes in position 2-3: truncated \\UXXXXXXXX escape\n\nPrinting out the relevant lines from the Psithon --> Python processed input file:\n\n\n!----------------------------------------------------------------------------------!\n!                                                                                  !\n!  (unicode error) \'unicodeescape\' codec can\'t decode bytes in position 2-3:       !\n!     truncated \\UXXXXXXXX escape                                                  !\n!                                                                                  !\n!----------------------------------------------------------------------------------!\n\n'
```

With the kind help of some psi4 developers I could track down it to these lines:
https://github.com/Qiskit/qiskit-nature/blob/c7fa055cfba2dc08daef0111e569ae5eb9bce10b/qiskit_nature/drivers/psi4d/psi4driver.py#L123-L133

I was able to get rid of the problem via changing above lines to this:
```python
        qiskit_nature_directory = os.path.abspath(os.path.join(psi4d_directory, "../.."))

        molecule = QMolecule()

        input_text = cfg + "\n"
        input_text += "import sys\n"
        syspath = "['" + qiskit_nature_directory.replace("\\", "\\\\") + "','" + "','".join(p.replace("\\", "\\\\") for p in sys.path) + "']"

        input_text += "sys.path = " + syspath + " + sys.path\n"
        input_text += "from qiskit_nature.drivers.qmolecule import QMolecule\n"
        input_text += "_q_molecule = QMolecule('" + molecule.filename.replace("\\", "\\\\") + "')\n"
```
Unfortunately this fix is not cross-platform compatible but quite windows-specific.
So, any help finding a nice fix for this bug or even a simplified/less error-prone way to call psi4 would be appreciated.

**Comment by cpossel** [comment]
Maybe the initial issue at psi4 is useful for you. It can be found here
[https://github.com/psi4/psi4/issues/2228](url)

**Comment by loriab** [comment]
Hi, I'm a psi4 dev. Please ping me some time if you'd like to switch the Qiskit interface to things like:
* `import psi4;ret, wfn = psi4.energy(mtd, return_wfn=True)` instead of `psi4 -i input_text.in`
* `psi4.core.Molecule.from_arrays(geom=[0, 1, 0, ..., 5.3], elem=["H", "O", "H"], molecular_charge=0)` instead of building from string.

**Comment by woodsp-ibm** [comment]
I know when we first did the PSI4 driver there was no Windows version and I guess it has not been looked at since that came out. As far as it being Windows specific since we can detect platform its possible to do different code paths if that's what it takes.

@loriab The reason we went with the input file approach was to allow users to program at that level with PSIThon and do settings etc like they were used to. It was part of the requirements we had at the time. I have thought about having a version that is more a direct Python programming interface but its something we have not done yet.

**Comment by loriab** [comment]
> The reason we went with the input file approach was to allow users to program at that level with PSIThon and do settings etc like they were used to. It was part of the requirements we had at the time. I have thought about having a version that is more a direct Python programming interface but its something we have not done yet.

That's perfectly understandable, and thanks for catering to our DSL. Nowadays, using the API isn't much harder than adding some `psi4.`s to Psithon and `molecule <name> {...}` to `name = psi4.geometry(...)` and `set {key val}` -> `psi4.set_options({"key": val})`. I usually think of quantum computing users as pretty adept, so a change may be reasonable to avert and trace errors like the above. Anyways, if you ever want to tackle it, give me a ping.

> I know when we first did the PSI4 driver there was no Windows version and I guess it has not been looked at since that came out. As far as it being Windows specific since we can detect platform its possible to do different code paths if that's what it takes.

Windows and other OS psi4 packages are available to CI through conda, if that's ever a help.

**Comment by mrossinek** [comment]
If the problem of this issue is related to paths I suggest looking into [pathlib](https://docs.python.org/3/library/pathlib.html) to replace the `os.path` usage. It has built-in platform support/detection and should simplify these things a lot.

**Comment by cpossel** [comment]
I added a pull request for my changes here:
[https://github.com/Qiskit/qiskit-nature/pull/262](url)
@mrossinek Thanks for pointing out to use pathlib.

**Comment by woodsp-ibm** [comment]
We should leave this open until the PR that fixes this #262 is merged. I edited it so once it merges it will auto-close this.
